### DISCUSSION TITLE
Stack overflow on a recursive query

### DISCUSSION THREAD

**Discussion by DGolubets** [apollographql/router#4426] [Q&A]
I can easily abort the router process by sending something like this:
```
query Recursion {
  organization {
    createdBy {
      ...UserFragment @defer
    }
  }
}

fragment UserFragment on User {
  organization {
    createdBy {
       ...UserFragment @defer
    }
  }
}
```

I though `parser_max_recursion` should protect against this but apparently it's not.
Any other options?
Should I just disable `@defer`?

**Comment by BrynCooke** [comment]
This looks like does fail to parse on apollo-rs main: 
```
Error: `UserFragment` fragment cannot reference itself
    ╭─[0041_recursive_fragment.graphql:24:1]
    │
 24 │ fragment UserFragment on User {
    │ ──────────┬──────────  
    │           ╰──────────── recursive fragment definition
    │ 
 27 │             ...UserFragment @defer
    │             ───────────┬──────────  
    │                        ╰──────────── `UserFragment` circularly references `UserFragment` here
────╯
```

Will check if this is still an issue on the latest version of the router.

**Reply by BrynCooke** [reply]
It does still occur. Gonna convert this into an issue.

**Comment by SimonSapin** [comment]
This was filed as https://github.com/apollographql/router/issues/4450 and fixed in https://github.com/apollographql/router/pull/4551
